core/src/main/scala/com/stripe/dagon/HMap.scala

```diff
@@ -80,6 +83,6 @@ object HMap {
   def empty[K[_], V[_]]: HMap[K, V] =
     from[K, V](Map.empty[K[_], V[_]])
 
-  private def from[K[_], V[_]](m: Map[K[_], V[_]]): HMap[K, V] =
+  def from[K[_], V[_]](m: Map[K[_], V[_]]): HMap[K, V] =
```

Comment:
do we have to make this public? This is unsafe and defeating the main use of the code (we know that the inner types match for the keys).

It looks like you are just using it to generate `HMap`s, but I think we can do that by doing `.update` with the right way, no?

Reply:
I'll make it `private[dagon]` -- I'm just using it in tests.